### ISSUE TITLE
iOS - VoiceOver Does Not Focus on Top of Screen When Going to New Location

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#94]
Why when a VoiceOver user goes to a new screen does the focus not go to the top of the screen which should be the ILAB logo and not the first heading on the screen or somewhere else.

An example would be when tapping the "Why Develop a Social Compliance System?" from the home screen or the Menu it focuses on what I would expect to be the heading and skips all the other things above it. Or if the user goes to one of the Steps, the focus goes to one of the collapsed sections depending on the Step or maybe which one I had open previously.

**Comment by jfc3-dol** [comment]
VoiceOver is still focusing on the text "Why Develop a Social Compliance System?" when I go to that screen instead of the top of the screen as described above.

**Comment by KyleLivingstonDOL** [comment]
@jfc3-dol Thanks for this. I am confused. When I click on "Why Develop a Social Compliance System" on my iPhone from either the home page or menu selection, I am brought to that specific page.

I do not see the "all the other things about it [it being the heading]". Can you please elaborate what you mean by this?

**Comment by jfc3-dol** [comment]
@KyleLivingstonDOL,

When I looked at the screen VoiceOver went to, I hadn't gone to read the text, but because it was bold and larger, I assumed it was a heading. Sorry about that confusion.

The main issue is that when a VoiceOver user goes to the screen after tapping a link, they are placed somewhere on the screen instead of at the top of the screen. So they are missing information that is above them.

If you want me to show you what I'm talking about, then we can set up a meeting in my office.

**Comment by KyleLivingstonDOL** [comment]
Thanks John. ZTI, can you please review and ensure voiceover automatically defaults to the top of the page only when going to a new page. When going back, please ensure voiceover defaults to the last clicked link so it does not start from the top again.

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

From our conversation a few days ago, I can see how React-Native could be a pain for iOS applications since it can't be all things to all devices and work as expected.

Once you choose a new screen, VoiceOver seems to focus on the new screen back to where you tapped on the previous screen or as close to it as possible instead of the top of the screen as expected.

Please see the attached screenshots showing where I was and where VoiceOver focused on the new screen.

**From Step 1 choose Topic 2.**

![ILAB Comply Chain - ios - for Step 1 I choose Topic 2](https://user-images.githubusercontent.com/10423487/91467744-fd274300-e85e-11ea-8076-32fe1505d6c5.png)

**Where VoiceOver focused after choosing Topic 2 from Step 1.**

![ILAB Comply Chain - ios - where VoiceOver focused after choosing Topic 2 from Step 1](https://user-images.githubusercontent.com/10423487/91467758-01536080-e85f-11ea-9d33-f3237333642c.png)

**Comment by harpreetsdev** [comment]
@jfc3-dol  Please test this Issue once again when you get a chance. The latest build (deployed yesterday) should be fine for testing. This issue is no longer showing up on my iPhone 12 Pro Max, the focus lands on the page top as expected. I feel subsequent changes in the page structures due to content changes over last few sprints has caused this. 

Please test Issue #102 as well. That one working as expected too.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

When I tested the "Why Develop a Social Compliance System?" from the home screen with VoiceOver, it highlights the middle of the screen, as shown in the attached screenshot.

![comply chain ios - why develop a social compliance system - VoiceOver starts reading where the black border is highlighting the text](https://user-images.githubusercontent.com/10423487/102369801-5b9d3c00-3f8a-11eb-8042-1e82993c9e35.png)

I'm not sure if it's fixed for the latest version of iOS only is why it is working for you or not for me. 

Our device is on iOS 13.3.1. 

I do not want to upgrade because if it has issues and gets bricked/does not work anymore; I don't have a device to test with then. 

In the past, I have had my personal iPhone and iPad not update properly, and I don't want to have to go to an Apple store to get it fixed during a pandemic. Plus, I'm not sure the upgrade will be able to be done on our older devices and the password to upgrade them is my boss's.

**Comment by jfc3-dol** [comment]
@pbhatt17 

VoiceOver is still starting to read in the middle of the page and not at the top as expected.

**Comment by rveruna** [comment]
I'm guessing this is caused by a known bug in react native navigation https://github.com/react-navigation/react-navigation/issues/7056

**Comment by KyleLivingstonDOL** [comment]
Adding James and Alex.

Kyle S. Livingston
Lead International Relations Officer (Senior Research Coordinator)
U.S. Department of Labor | Bureau of International Labor Affairs
Office of Child Labor, Forced Labor, and Human Trafficking
***@***.******@***.***> | Teleworking: 503-720-4105
he/him/his
